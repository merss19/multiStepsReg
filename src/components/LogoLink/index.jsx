"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_router_dom_1 = require("react-router-dom");
var storage_1 = require("tools/storage");
var LogoLink = function () { return (<react_router_dom_1.Link to="/" onClick={function () {
    storage_1.storage.remove('token', { path: '/' });
    storage_1.storage.remove('program', { path: '/' });
    storage_1.storage.remove('promoName', { path: '/' });
}}>
    <svg className='svg-icon ys-logo-web'>
      <use xlinkHref='#ys-logo-web'/>
    </svg>
  </react_router_dom_1.Link>); };
exports.default = LogoLink;
