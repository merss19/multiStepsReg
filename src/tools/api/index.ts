const api = 'https://api.todayme.ru/api'

export const apiData = {
  sendRestorePassword: `${api}/user/user-sendRestorePassword`,
  approveRestorePassword: `${api}/user/user-approveRestorePassword`,
  authenticate: `${api}/user/authenticate`,
  userCreate: `${api}/user/user-create`,
  userCheck: `${api}/user/user-check`,
  userGet: `${api}/user/user-get`,
  packageGet: `${api}/day/package-get`,
  userUpdate: `${api}/user/user-update`,
  paymentCreate: `${api}/payment/payment-create`,
  paymentGetInfo: `${api}/payment/payment-get-info`,
  paymentSetpaidManual: `${api}/payment/payment-setpaid-manual`,
  programGet: `${api}/day/program-get`,
  authenticateSocial: `${api}/user/authenticate-social`
};

export function callApi(api: Promise<any>): any {
  return api
    .then(response => {
        return response.json().then((json: any) => {
            return {json, response}
          }
        )
      }
    ).then(({json, response}) => {
      if (response.status !== 200) {
        return Promise.reject(response)
      }
      return json
    })
    .then(
      json => {
        return json
      },
      error => {
        return {error: error.message || 'Возникла ошибка'}
      }
    )
}
