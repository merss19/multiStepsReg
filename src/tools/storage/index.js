"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var cookie = require("react-cookie");
exports.storage = {
    save: function (name, data, path) {
        cookie.save(name, data, path);
    },
    load: function (name) {
        return cookie.load(name);
    },
    remove: function (name, path) {
        return cookie.remove(name, path);
    },
    loadState: function (name) {
        try {
            var serializedState = localStorage.getItem(name);
            if (!serializedState) {
                return undefined;
            }
            return JSON.parse(serializedState);
        }
        catch (err) {
            return undefined;
        }
    },
    saveState: function (data, name) {
        try {
            var serializedState = JSON.stringify(data);
            localStorage.setItem(name, serializedState);
        }
        catch (err) {
            //console.log(err)
        }
    },
    removeState: function (name) {
        try {
            localStorage.removeItem(name);
        }
        catch (err) {
            ///console.log(err)
        }
    }
};
