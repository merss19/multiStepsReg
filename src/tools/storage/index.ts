import * as cookie from 'react-cookie';

interface StorageAction {
	save: (name: string, data: any, path?: object) => void
	load: (name: string) => any
	remove: (name: string, path?: object) => void
	loadState: (name: string) => any,
	saveState: (data: any, name: string) => void
	removeState: (name: string) => void
}
export const storage: StorageAction = {
  save: (name: string, data: any, path?: object): void => {
    cookie.save(name, data, path)
  },
  load: (name: string): any => {
    return cookie.load(name)
  },
  remove: (name: string, path?: object): any => {
    return cookie.remove(name, path)
  },
  loadState: (name: string) => {
    try {
      const serializedState = localStorage.getItem(name);
      if (!serializedState) {
        return undefined
      }
      return JSON.parse(serializedState);
    } catch (err) {
      return undefined
    }
  },
  saveState: (data: any, name: string) => {
    try {
      const serializedState = JSON.stringify(data);
      localStorage.setItem(name, serializedState)
    } catch (err) {
      //console.log(err)
    }
  },
  removeState: (name: string) => {
    try {
      localStorage.removeItem(name)
    } catch (err) {
      ///console.log(err)
    }
  }
};