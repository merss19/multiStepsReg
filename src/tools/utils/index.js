"use strict";
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
function makeActionCreator(type) {
    var argNames = [];
    for (var _i = 1; _i < arguments.length; _i++) {
        argNames[_i - 1] = arguments[_i];
    }
    return function () {
        var args = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            args[_i] = arguments[_i];
        }
        var action = { type: type };
        argNames.forEach(function (arg, index) {
            if (arg === 'data') {
                action[argNames[index]] = args[index];
            }
            else if (arg === 'action') {
                var addType = args[index];
                if (addType) {
                    var addTypeArr = addType.split('/');
                    action.type = type + '/' + addTypeArr[addTypeArr.length - 1];
                }
            }
            else {
                action[argNames[index]] = args[index];
            }
        });
        return action;
    };
}
exports.makeActionCreator = makeActionCreator;
exports.addActiveFlag = function (items) {
    items = items.map(function (item) {
        return __assign({}, item, { isActive: false });
    });
    return items;
};
exports.queryParse = function (search) {
    // console.log('queryParse')
    //console.log(search)
    if (!search) {
        return null;
    }
    var parsed = search.slice(1).split('&');
    var query = {};
    parsed.forEach(function (item) {
        var arr = item.split('=');
        if (!arr.length) {
            return query;
        }
        //console.log(arr)
        // console.log(arr[0])
        //console.log(arr[1])
        query[arr[0]] = arr[1];
    });
    //console.log(parsed)
    //console.log(query)
    //console.log('queryParse')
    return query;
};
exports.isInEnum = function (en, el) {
    // console.log('isInEnum');
    var result = false;
    for (var item in en) {
        if (isNaN(Number(item))) {
            // console.log(el);
            //console.log(en[item]);
            //console.log(en[item].sort);
            if (en[item] == el) {
                result = true;
            }
        }
    }
    //console.log('result');
    //console.log(result);
    return result;
};
exports.localStorageMock = (function () {
    var store = {};
    return {
        getItem: function (key) {
            return store[key];
        },
        setItem: function (key, value) {
            store[key] = value.toString();
        },
        clear: function () {
            store = {};
        },
        removeItem: function (key) {
            delete store[key];
        }
    };
})();
