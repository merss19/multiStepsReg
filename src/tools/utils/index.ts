
export function makeActionCreator<T>(type:string, ...argNames: string[]){
  return (...args: any[]) => {
    let action= { type: type };
    argNames.forEach((arg, index) => {
      if(arg === 'data'){
        action[argNames[index]] = <T>args[index]
      } else if (arg === 'action'){
        const addType:string = args[index]
        if(addType){
          const addTypeArr = addType.split('/');
          action.type = type + '/' + addTypeArr[addTypeArr.length - 1]
        }
      } else {
        action[argNames[index]] = <any>args[index]
      }
    });
    return action;
  }
}

export const addActiveFlag = (items: any[]) => {
  items = items.map((item: any) => {
    return {
      ...item,
      isActive: false
    }
  });
  return items
}

export const queryParse = (search:string) => {
 // console.log('queryParse')
  //console.log(search)
  if(!search){
    return null;
  }
  const parsed = search.slice(1).split('&');
  let query = {};
  parsed.forEach((item:any) => {
    let arr = item.split('=');
    if(!arr.length){
      return query;
    }
    //console.log(arr)
  // console.log(arr[0])
    //console.log(arr[1])
    query[arr[0]] = arr[1]
  });
  //console.log(parsed)
  //console.log(query)
 //console.log('queryParse')
  return query;
};

export const isInEnum = (en: any, el: string | number) =>{
 // console.log('isInEnum');
  let result: boolean = false
  for (let item in en) {
    if (isNaN(Number(item))) {
     // console.log(el);
      //console.log(en[item]);
      //console.log(en[item].sort);
      if(en[item] == el){
        result = true;
      }
    }
  }
  //console.log('result');
  //console.log(result);
  return result;
};

export let localStorageMock = (function () {
  let store = {};
  return {
    getItem: function (key: any) {
      return store[key];
    },
    setItem: function (key: any, value: any) {
      store[key] = value.toString();
    },
    clear: function () {
      store = {};
    },
    removeItem: function (key: any) {
      delete store[key];
    }
  };
})();
