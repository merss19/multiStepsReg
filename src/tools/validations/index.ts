
const isEmpty = (value:any) => value === undefined || value === null || value === '';
const join = (rules:any) => {
  return (value:any) => {
    return rules.map((rule:any) => {
      return rule(value)
    }).filter((error:any) => {
      return !!error
    })[0 /* first error */ ];
  }
};

export const validateGender = (value: any) => {
  if(!value) {
    return 'Поле пол должно быть заполнено'
  }
};
const validateEmail = (data:any, errors: any) => {
  if (data.email) {
    data.email = data.email.replace(/ /g, '');
  }
  if (!data.email) {
    errors.email = 'Email должен быть заполнен';
  }
  if (!/^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/i.test(data.email)) {
    errors.email = 'Email заполнен неправильно';
  }
  return errors;
}

const validatePassword = (data:any, errors: any) => {
  if (!data.password) {
    errors.password = 'Поле пароля должно быть заполнено';
  }
  if (data.password && data.password.length < 6) {
    errors.password = 'Поле пароля должно быть длиннее 6 символов';
  }
  if (data.password && data.password.length > 20) {
    errors.password = 'Поле пароля должно быть короче 20 символов';
  }
  if (/["]/g.test(data.password)) {
    errors.password = 'Поле пароля не должно содержать знак "';
  }

  if (data.password !== data.passwordAgain) {
    errors.passwordAgain = 'Пароли должны совпадать';
  }
  return errors;
}

export const validateForgetPassword = (data: any) => {
  let errors: any = {};
      errors = validateEmail(data, errors);
  return errors;
};

export const validatePasswordRestore = (data: any) => {
  let errors: any = {};
      errors = validatePassword(data, errors);
  return errors;
};
export const validateSignUp = (data: any) => {
  let errors: any = {};
      errors = validateEmail(data, errors);
      errors = validatePassword(data, errors);

  if (!data.gender) {
    errors.gender = 'Пол должен быть заполнен';
  }

  if (!data.accept) {
    errors.accept = 'Вы должны принять условия оферты и правил';
  }
  return errors;
};

export const validateLogin = (data: any) => {
  let errors: any = {};
      errors = validateEmail(data, errors);
  if (!data.password) {
    errors.password = 'Поле пароля должно быть заполнено';
  }
  return errors;
};

export function getBsFeedback(value:any) {
  var feedback;
  feedback = {};
  if (value.touched) {
    feedback = {
      hasFeedback: true,
      bsStyle: value.valid ? 'success' : 'error',
      help: value.touched && (value.error ? value.error : '')
    };
  }
  return feedback;
}

export function email(value:any) {
  if (!isEmpty(value) && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value)) {
    return 'Invalid email address';
  }
}

export function required(value:any) {
  if (isEmpty(value)) {
    return 'Required';
  }
  return false;
}

export function minLength(min:any) {
  return (value:any) => {
    if (!isEmpty(value) && value.length < min) {
      return `Must be at least ${min} characters`;
    }
    return false;
  };
}

export function maxLength(max:any) {
  return (value:any) => {
    if (!isEmpty(value) && value.length > max) {
      return `Must be no more than ${max} characters`;
    }
    return false;
  };
}

export function integer(value:any) {
  if (!Number.isInteger(Number(value))) {
    return 'Must be an integer';
  }
  return false;
}

export function oneOf(enumeration:any) {
  return (value:any) => {
    if (!~enumeration.indexOf(value)) {
      return `Must be one of: ${enumeration.join(', ')}`;
    }
    return false;
  };
}

export function match(field:any) {
  return (value:any, data:any) => {
    if (data) {
      if (value !== data[field]) {
        return 'Do not match';
      }
    }
    return false;
  };
}


export function createValidator(rules: any) {
  return (data = {}) => {
    const errors = {};
    Object.keys(rules).forEach((key) => {
      const rule = join([].concat(rules[key])); // concat enables both functions and arrays of functions
      const error = rule(data[key]);
      if (error) {
        errors[key] = error;
      }
    });
    return errors;
  };
}
