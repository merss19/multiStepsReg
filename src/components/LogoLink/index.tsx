import * as React from 'react';
import { Link } from 'react-router-dom';
import {storage} from 'tools/storage';

const LogoLink: React.SFC<{}> = () => (
  <Link
    to="/"
    onClick={() => {
      storage.remove('token', {path: '/'});
      storage.remove('program', {path: '/'});
      storage.remove('promoName', {path: '/'});
    }}
  >
    <svg className='svg-icon ys-logo-web'>
      <use xlinkHref='#ys-logo-web'/>
    </svg>
  </Link>
);

export default LogoLink;
