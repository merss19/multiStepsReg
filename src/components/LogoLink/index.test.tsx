import {shallow, mount} from 'enzyme';
import LogoLink from './index';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {storage} from '../../tools/storage';

configure({ adapter: new Adapter() });
storage.save('token', 'testToken',
	{ path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
describe('Component Loader', () => {
	const component = shallow(<LogoLink />);
	it('renders component', () => {
		expect(component).toHaveLength(1);
	});
	it('renders correctly', () => {
		const tree = toJson(shallow(<LogoLink />));
		expect(tree).toMatchSnapshot();
	});
	it('simulate click', () => {
		const token = storage.load('token');
		expect(token).toBe('testToken');
		component.find('Link').simulate('click');
		const tokenRemove = storage.load('token');
		expect(tokenRemove).toBe(undefined);
	});
});