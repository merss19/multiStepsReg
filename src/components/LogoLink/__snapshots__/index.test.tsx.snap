// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Loader renders correctly 1`] = `
<Link
  onClick={[Function]}
  replace={false}
  to="/"
>
  <svg
    className="svg-icon ys-logo-web"
  >
    <use
      xlinkHref="#ys-logo-web"
    />
  </svg>
</Link>
`;
